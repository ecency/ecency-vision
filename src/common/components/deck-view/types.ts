export interface DeckModel {
  data: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
  };
  listItemComponent: any;
  createdAt: Date;
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}
