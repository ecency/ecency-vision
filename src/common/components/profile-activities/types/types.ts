export interface ActivityTypes {
    type: string;
      parent_author?: string;
      parent_permlink?: string;
      author?: string;
      permlink?: string;
      payment?: number;
      payer?: string;
      following?: string;
      community?: string;
      approve?: boolean;
      proposal_ids?: string;
      account?: string;
      id?: string;
      what?: string[];
      voter?: string;
      witness?: string;
      json: string;
    timestamp: string;
  }