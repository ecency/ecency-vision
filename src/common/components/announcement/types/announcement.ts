export interface Announcement {
  id: number;
  title: string;
  description: string;
  button_text: string;
  button_link: string;
}

export interface LaterAnnouncement {
  id: number;
  dateTime: Date;
}
