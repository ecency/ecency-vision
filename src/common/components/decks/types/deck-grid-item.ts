interface BaseDeckGridItem {
  id: string;
  key: number;
  type: "u" | "ac" | "co" | "w" | "n" | "tr" | "to" | "s" | "cu" | "th";
  settings: Record<string, any>;
}

export interface ReloadableDeckGridItem extends BaseDeckGridItem {
  settings: {
    updateIntervalMs: number;
  };
}

export interface UserDeckGridItem extends ReloadableDeckGridItem {
  type: "u";
  settings: ReloadableDeckGridItem["settings"] & {
    username: string;
    contentType: string;
  };
}

export interface CommunityDeckGridItem extends ReloadableDeckGridItem {
  type: "co";
  settings: UserDeckGridItem["settings"] & {
    tag: string;
  };
}

export interface SearchDeckGridItem extends ReloadableDeckGridItem {
  type: "s";
  settings: ReloadableDeckGridItem["settings"] & {
    query: string;
  };
}

export interface BitesDeckGridItem extends ReloadableDeckGridItem {
  type: "th";
  settings: ReloadableDeckGridItem["settings"] & {
    host: string;
  };
}

export type DeckGridItem =
  | BaseDeckGridItem
  | UserDeckGridItem
  | CommunityDeckGridItem
  | SearchDeckGridItem
  | BitesDeckGridItem;
