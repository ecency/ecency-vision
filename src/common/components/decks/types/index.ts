export * from "./deck-grid-item";
export * from "./deck-grid";
export * from "./old-decks";
