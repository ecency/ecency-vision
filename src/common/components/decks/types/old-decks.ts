interface GenericDeck<DF extends Record<string, string | number> | null> {
  createdAt: string;
  dataFilters: DF;
  dataParams: string[];
  header: {
    title: string;
    updateIntervalMs: number;
  };
}

export interface OldNotificationDeck extends GenericDeck<{ type: string; group: string }> {}
export interface OldNonDFDeck extends GenericDeck<null> {}

export type OldDeck = OldNotificationDeck | OldNonDFDeck;
