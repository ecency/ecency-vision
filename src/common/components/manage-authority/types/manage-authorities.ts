import { Account, FullAccount } from "../../../store/accounts/types";
import { PublicKey } from "@hiveio/dhive";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting",
  Memo: "memo"
};

export interface PublicKeys {
  publicOwnerKey?: string;
  publicActiveKey?: string;
  publicPostingKey?: string;
  publicMemoKey?: string;
}

export interface AccountDataType {
  postingsAuthority: [string, number][];
  posting: [string | PublicKey, number];
  owner: [string | PublicKey, number];
  active: [string | PublicKey, number];
  weight: number;
  memokey: string;
  account: Account;
  publicKeys: {
    publicOwnerKey: string | PublicKey;
    publicActiveKey: string | PublicKey;
    publicPostingKey: string | PublicKey;
    publicMemoKey: string;
  };
}

export const actionType = {
  Revoke: "revoke",
  Keys: "keys",
  Import: "import",
  Reveal: "reveal"
};
