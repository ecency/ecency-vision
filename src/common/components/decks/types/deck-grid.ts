import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  key: string;
  title: string;
  icon?: string;
  columns: DeckGridItem[];
  storageType: "account" | "local";
}

export interface DeckGrids {
  decks: DeckGrid[];
}
