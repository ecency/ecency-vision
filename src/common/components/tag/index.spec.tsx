import React from 'react';

import Tag from './index';
import TestRenderer from 'react-test-renderer';
import { createBrowserHistory } from 'history';

import { globalInstance, communityInstance1 } from '../../helper/test-helper';

import { EntryFilter, AllFilter } from '../../store/global/types';

jest.mock('../../api/bridge', () => ({
  getCommunity: () =>
    new Promise((resolve) => {
      resolve(communityInstance1);
    })
}));

it('(1) Link', () => {
  const props = {
    history: createBrowserHistory(),
    global: {
      ...globalInstance,
      ...{
        filter: EntryFilter.hot,
        tag: 'bitcoin'
      }
    },
    tag: 'bitcoin'
  };

  const renderer = TestRenderer.create(
    <Tag {...props} type="link">
      <span>bitcoin</span>
    </Tag>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});

it('(2) Span', () => {
  const props = {
    history: createBrowserHistory(),
    global: {
      ...globalInstance,
      ...{
        filter: EntryFilter.hot,
        tag: 'bitcoin'
      }
    },
    tag: 'bitcoin'
  };

  const renderer = TestRenderer.create(
    <Tag {...props} type="span">
      <span>bitcoin</span>
    </Tag>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});

it('(3) Community as Link', () => {
  const props = {
    history: createBrowserHistory(),
    global: {
      ...globalInstance,
      ...{
        filter: EntryFilter.hot,
        tag: 'bitcoin'
      }
    },
    tag: 'hive-2321'
  };

  const renderer = TestRenderer.create(
    <Tag {...props} type="span">
      <span>hive-2321</span>
    </Tag>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(4) Should use default filter if the globl filter is 'feed'", () => {
  const props = {
    history: createBrowserHistory(),
    global: {
      ...globalInstance,
      ...{
        filter: AllFilter.feed,
        tag: 'bitcoin'
      }
    },
    tag: 'bitcoin'
  };

  const renderer = TestRenderer.create(
    <Tag {...props} type="link">
      <span>bitcoin</span>
    </Tag>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});

it('(5) Link with community tag', () => {
  const props = {
    history: createBrowserHistory(),
    global: {
      ...globalInstance
    },
    tag: { name: 'hive-125125', title: 'Ecency' }
  };

  const renderer = TestRenderer.create(
    <Tag {...props} type="link">
      <span>Ecency</span>
    </Tag>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});
