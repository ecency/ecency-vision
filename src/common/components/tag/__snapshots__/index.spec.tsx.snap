// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Link 1`] = `
<a
  href="/hot/bitcoin"
  onClick={[Function]}
  title="Unmoderated topics"
>
  bitcoin
</a>
`;

exports[`(2) Span 1`] = `
<span
  title="Unmoderated topics"
>
  bitcoin
</span>
`;

exports[`(3) Community as Link 1`] = `
<span
  title="Unmoderated topics"
>
  hive-2321
</span>
`;

exports[`(4) Should use default filter if the globl filter is 'feed' 1`] = `
<a
  href="/created/bitcoin"
  onClick={[Function]}
  title="Unmoderated topics"
>
  bitcoin
</a>
`;

exports[`(5) Link with community tag 1`] = `
<a
  href="/created/hive-125125"
  onClick={[Function]}
  title="Moderated community"
>
  Ecency
</a>
`;
